enterprise/backend/src/metabase_enterprise/library/api.clj

```diff
@@ -0,0 +1,84 @@
+(ns metabase-enterprise.library.api
+  (:require
+   [metabase.api.common :as api]
+   [metabase.api.macros :as api.macros]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.collections.core :as collections]
+   [metabase.collections.models.collection :as collection]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(api.macros/defendpoint :post "/"
+  "Creates the Library if it doesn't exist. Returns the created collection.
+
+  Requires superuser permissions."
+  [_route
+   _query
+   _body]
+  (api/check-superuser)
+  (api/check-400 (not (collections/library-collection)) "Library already exists")
```

Comment:
Maybe it doesn't need to error it could just return 204?

Reply:
I figured a 400 would make more since becase it's not doing what you are expecting it to and not going to return what you expect. The FE shouldn't be calling this if the library exists ever, so it's more of a fallback catch for it doing something weird so good to make that more explicit.